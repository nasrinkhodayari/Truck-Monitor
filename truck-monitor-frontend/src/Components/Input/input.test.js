import React from "react";
import { render, shallow } from "enzyme";
import AppInputField from "./index";

describe("App TextInput component", () => {
  const maxLength = 6,
    value = 'AALL44',
    type = 'text',
    placeholder = 'license plate',
    changeHandler = jest.fn();
  let InputFieldComponent;
  beforeEach(() => {
    InputFieldComponent = (
      <AppInputField
        value={value}
        maxLength={maxLength}
        type={type}
        placeholder={placeholder}
        changeHandler={changeHandler}
      />);
  });
  it("TextInput render correctly (Match Snapshot)", () => {
    const wrapper = render(InputFieldComponent);
    expect(wrapper).toMatchSnapshot();
  });

  it("TextInput OnChange calles correctly", () => {
    const wrapper = shallow(InputFieldComponent);
    wrapper.simulate('change', { target: { value } });
    expect(changeHandler).toBeCalledWith({ target: { value } });
  });

});