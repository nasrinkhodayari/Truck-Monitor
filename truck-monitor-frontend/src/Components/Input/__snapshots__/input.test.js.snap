// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App TextInput component TextInput render correctly (Match Snapshot) 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "app-input",
      "maxlength": "6",
      "placeholder": "license plate",
      "type": "text",
      "value": "AALL44",
    },
    "children": Array [],
    "name": "input",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "maxlength": undefined,
      "placeholder": undefined,
      "type": undefined,
      "value": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "maxlength": undefined,
      "placeholder": undefined,
      "type": undefined,
      "value": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
